query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresBasedRequestHandler.java

```diff
@@ -478,77 +458,11 @@ && isEquals(tableColumnFilter.getValues(), queryFilter.getRhs())) {
     return queryFilter;
   }
 
-  Observable<Row> convert(ResultSet resultSet, LinkedHashSet<String> selectedAttributes)
-      throws SQLException {
+  Observable<Row> convert(ResultSet resultSet) throws SQLException {
     List<Builder> rowBuilderList = new ArrayList<>();
-    if (resultSet.next()) {
-      // Postgres has different Response format for selection and aggregation/group by query.
-      if (resultSetTypePredicateProvider.isSelectionResultSetType(resultSet)) {
-        // map merging is only supported in the selection. Filtering and Group by has its own
-        // syntax in Postgres
-        handleSelection(resultSet, rowBuilderList, selectedAttributes);
-      } else if (resultSetTypePredicateProvider.isResultTableResultSetType(resultSet)) {
-        handleTableFormatResultSet(resultSet, rowBuilderList);
-      } else {
-        handleAggregationAndGroupBy(resultSet, rowBuilderList);
-      }
-    }
-    return Observable.fromIterable(rowBuilderList)
-        .map(Builder::build)
-        .doOnNext(row -> LOG.debug("collect a row: {}", row));
-  }
-
-  private void handleSelection(
-      ResultSet resultSet, List<Builder> rowBuilderList, LinkedHashSet<String> selectedAttributes)
-      throws SQLException {
-    do {
-      // Find the index in the result's column for each selected attributes
-      PostgresResultAnalyzer resultAnalyzer =
-          PostgresResultAnalyzer.create(resultSet, selectedAttributes, tableDefinition);
-      Builder builder;
-      builder = Row.newBuilder();
+    while (resultSet.next()) {
```

Comment:
Just a Note: Looks fine, just verify once that column order of the result set matches with the column order returned in query-request response metadata (As I remember, it's the same selection order, but let's cross-check once).

Reply:
I have validated that part